tests/console/validate_ext4_fs.pm

```diff
@@ -0,0 +1,91 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate ext4 filesystem and system partitioning.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+
+    select_console "root-console";
+
+    my $test_data  = get_test_suite_data();
+    my @partitions = @{$test_data->{partitions}};
+    my $size       = 0;
+    my $tmp;
+
+    foreach (@partitions) {
+
+        if ($_->{mnt_point} eq '[SWAP]') {
```

Comment:
you can exclude with if the block to test ability to read and write files if it is swap, but still check fs_type for swap.